std/socket.d

```diff
@@ -1269,6 +1269,14 @@ abstract class Address
     /// Returns actual size of underlying $(D sockaddr) structure.
     abstract @property socklen_t nameLen() const pure nothrow @nogc;
 
+    // Sets the actual size of underlying $(D sockaddr).
+    // Kludge for setting path length in UnixAddress
+    protected @property void nameLen(socklen_t len) pure
+    {
+        if (len > this.nameLen)
+            throw new SocketParameterException("Not enough socket address storage");
+    }
+
```

Comment:
I don't understand the point of this method. The base `Address` class does not own the storage for the address data, so where the address itself is stored or how it deals with its storage/length is up to the descendant classes.

Setting the length for an address of fixed length (e.g. IP addresses) is nonsensical, so I don't see why this should be part of the abstract Address interface.

Furthermore, it seems to break the expectation that invoking a property's setter with some value should either throw or cause invoking the getter right afterwards to return the same value.